cookbook/workflows/async/blog_post_generator.py

```diff
@@ -0,0 +1,487 @@
+        print(f"   {i}. {article.title[:60]}...")
+
+    # Phase 2: Content extraction
+    print(f"\n📄 PHASE 2: CONTENT EXTRACTION")
+    print("=" * 50)
+
+    scraped_articles = await scrape_articles(
+        workflow, blog_topic, search_results, use_scrape_cache
+    )
+
+    if not scraped_articles:
+        return f"❌ Could not extract content from any articles for topic: {blog_topic}"
+
+    print(f"📖 Successfully extracted content from {len(scraped_articles)} articles")
+
+    # Phase 3: Blog post writing
+    print(f"\n✍️ PHASE 3: BLOG POST CREATION")
+    print("=" * 50)
+
+    # Prepare input for the writer
+    writer_input = {
+        "topic": blog_topic,
+        "articles": [article.model_dump() for article in scraped_articles.values()],
+    }
+
+    print("🤖 AI is crafting your blog post...")
+    writer_response = await blog_writer_agent.arun(json.dumps(writer_input, indent=2))
+
+    if not writer_response or not writer_response.content:
+        return f"❌ Failed to generate blog post for topic: {blog_topic}"
+
+    blog_post = writer_response.content
+
+    # Cache the blog post
+    cache_blog_post(workflow, blog_topic, blog_post)
+
+    print("✅ Blog post generated successfully!")
+    print(f"📝 Length: {len(blog_post)} characters")
+    print(f"📚 Sources: {len(scraped_articles)} articles")
+
+    return blog_post
+
+
+# --- Workflow Definition ---
+blog_generator_workflow = Workflow(
+    name="Blog Post Generator v2.0",
+    description="Advanced blog post generator with research and content creation capabilities",
+    storage=SqliteStorage(
+        table_name="blog_generator_v2",
+        db_file="tmp/blog_generator_v2.db",
+        mode="workflow_v2",
+    ),
+    steps=blog_generation_execution,
+    workflow_session_state={},  # Initialize empty session state for caching
+)
+
+
+if __name__ == "__main__":
+    import random
+
+    # Fun example topics to showcase the generator's versatility
+    example_topics = [
+        "The Rise of Artificial General Intelligence: Latest Breakthroughs",
+        "How Quantum Computing is Revolutionizing Cybersecurity",
+        "Sustainable Living in 2024: Practical Tips for Reducing Carbon Footprint",
+        "The Future of Work: AI and Human Collaboration",
+        "Space Tourism: From Science Fiction to Reality",
+        "Mindfulness and Mental Health in the Digital Age",
+        "The Evolution of Electric Vehicles: Current State and Future Trends",
+        "Why Cats Secretly Run the Internet",
+        "The Science Behind Why Pizza Tastes Better at 2 AM",
+        "How Rubber Ducks Revolutionized Software Development",
+    ]
+
+    # Test with a random topic
+    topic = random.choice(example_topics)
+
+    print("🧪 Testing Blog Post Generator v2.0")
+    print("=" * 60)
+    print(f"📝 Topic: {topic}")
+    print()
+
+    # Generate the blog post
+    asyncio.run(
+        blog_generator_workflow.aprint_response(
```

Comment:
Lets do `arun` and pprint the run response (Check that pprint_run_response works). That will just fit slightly better.